File: common/src/main/java/com/ssblur/scriptor/item/bound/BoundAxe.java
package com.ssblur.scriptor.item.bound;

import com.ssblur.scriptor.color.CustomColors;
import dev.architectury.platform.Platform;
import dev.architectury.registry.client.rendering.ColorHandlerRegistry;
import net.fabricmc.api.EnvType;
import net.minecraft.world.item.AxeItem;
import net.minecraft.world.item.Tier;

public class BoundAxe extends AxeItem {
  public BoundAxe(Tier tier, Properties properties) {
    super(tier, properties);

    if(Platform.getEnv() == EnvType.CLIENT)
      ColorHandlerRegistry.registerItemColors((itemStack, t) -> t == 1 ? CustomColors.INSTANCE.getColor(itemStack) : 0xFFFFFFFF, this);
  }

}

File: common/src/main/kotlin/com/ssblur/scriptor/item/tools/bound/BoundAxe.kt
package com.ssblur.scriptor.item.tools.bound

import com.ssblur.scriptor.color.CustomColors.getColor
import com.ssblur.unfocused.helper.ColorHelper.registerColor
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.world.item.AxeItem
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Tier

class BoundAxe(tier: Tier, properties: Properties) : AxeItem(tier, properties) {
    init {
        try{ clientInit() } catch (_: NoSuchMethodError) {}
    }

    @Environment(EnvType.CLIENT)
    fun clientInit() {
        this.registerColor{ itemStack: ItemStack?, t: Int ->
            if (t == 1) getColor(
                itemStack!!
            ) else -0x1
        }
    }
}
